src/context-parser.js

```diff
@@ -180,6 +180,14 @@ FastParser.prototype.walk = function(i, input, endsWithEOF) {
             if(this.tags[0].toLowerCase() === this.tags[1].toLowerCase()) {
                 reconsume = 0;  /* see 12.2.4.13 - switch state for the following case, otherwise, reconsume. */
                 this.matchEndTagWithStartTag(symbol);
+                /* 
+                  After matchEndTagWithStartTag (with start tag name == end tag name), 
+                  the state will be transition back to DATA. 
```

Comment:
`STATE_BEFORE_ATTRIBUTE_NAME` and `STATE_SELF_CLOSING_START_TAG` would only lead to `STATE_DATA` when an end tag is created.